crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -0,0 +1,403 @@
+    let mut bridge_health = BridgeHealth::new(name, health_topic, &target);
+
+    loop {
+        let res = recv_event_loop.poll().await;
+        bridge_health.update(&res, &mut companion_bridge_half).await;
+
+        let notification = match res {
+            Ok(notification) => notification,
+            Err(_) => continue,
+        };
+
+        match notification {
+            // Forward messages from event loop to target
+            Event::Incoming(Incoming::Publish(publish)) => {
+                target
+                    .publish(
+                        transform_topic(&publish.topic),
+                        publish.qos,
+                        publish.retain,
+                        publish.payload.clone(),
+                    )
+                    .await
+                    .unwrap();
+                let publish = (publish.qos > QoS::AtMostOnce).then_some(publish);
+                companion_bridge_half.send(publish).await.unwrap();
+            }
+
+            // Forward acks from event loop to target
+            Event::Incoming(
+                Incoming::PubAck(PubAck { pkid: ack_pkid })
+                | Incoming::PubRec(PubRec { pkid: ack_pkid }),
+            ) => {
+                if let Some(msg) = forward_pkid_to_received_msg.remove(&ack_pkid) {
+                    target.ack(&msg).await.unwrap();
+                }
+            }
+
+            // Keep track of packet IDs so we can acknowledge messages
+            Event::Outgoing(Outgoing::Publish(pkid)) => match companion_bridge_half.recv().await {
+                // A message was forwarded by the other bridge half, note the packet id
+                Some(Some(msg)) => {
+                    forward_pkid_to_received_msg.insert(pkid, msg);
+                }
+
+                // A healthcheck message was published, ignore this packet id
+                Some(None) => {}
+
+                // The other bridge half has disconnected, break the loop and shut down the bridge
+                None => break,
+            },
+            _ => {}
+        }
+    }
+}
+
+type NotificationRes = Result<Event, ConnectionError>;
+
+struct BridgeHealth<'a> {
+    name: &'static str,
+    health_topic: Option<String>,
+    target: &'a AsyncClient,
+    last_err: Option<String>,
+}
+
+impl<'a> BridgeHealth<'a> {
+    fn new(name: &'static str, health_topic: Option<String>, target: &'a AsyncClient) -> Self {
+        Self {
+            name,
+            health_topic,
+            target,
+            last_err: Some("dummy error".into()),
+        }
+    }
+    async fn update(
+        &mut self,
+        result: &NotificationRes,
+        companion_bridge_half: &mut BidirectionalChannelHalf<Option<Publish>>,
+    ) {
+        let name = self.name;
+        let (err, health_payload) = match result {
+            Ok(_) => {
+                info!("MQTT bridge connected to {name} broker");
+                (None, MQTT_BRIDGE_UP_PAYLOAD)
```

Comment:
> It would make sense to use a JSON format message (although that sounds like a breaking API change to me)

Since the health endpoint service name itself is new (`tedge-mapper-bridge-c8y` and not `mosquitto-c8y-bridge`), it should be fine.

Reply:
>> It would make sense to use a JSON format message (although that sounds like a breaking API change to me)

> Since the health endpoint service name itself is new (`tedge-mapper-bridge-c8y` and not `mosquitto-c8y-bridge`), it should be fine.

Beware that the logic to differentiate a bridge related topic is really weak: 
https://github.com/thin-edge/thin-edge.io/blob/main/crates/extensions/c8y_mapper_ext/src/service_monitor.rs#L29